component/attr_cache/cacheMap_test.go

```diff
@@ -0,0 +1,377 @@
+   SOFTWARE
+*/
+
+package attr_cache
+
+import (
+	"container/list"
+	"testing"
+	"time"
+
+	"github.com/Seagate/cloudfuse/internal"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+// look at s3wrappers tests for reference. look at common folder tests as well for reference. half way between s3wrappers and utils test
+
+type cacheMapTestSuite struct {
+	suite.Suite
+	assert            *assert.Assertions
+	rootAttrCacheItem attrCacheItem
+}
+
+// what is every test going to need to test with?
+func (suite *cacheMapTestSuite) SetupTest() {
+	suite.assert = assert.New(suite.T())
+
+	rootAttr := internal.CreateObjAttrDir("")
+	suite.rootAttrCacheItem = *newAttrCacheItem(rootAttr, true, time.Now())
+
+	//set up nested Dir tree
+	nestedDir, nestedFiles := generateNestedDirectory("a")
+
+	for dir := nestedDir.Front(); dir != nil; dir = dir.Next() {
+		attr := internal.CreateObjAttrDir(dir.Value.(string))
+		suite.rootAttrCacheItem.insert(attr, true, time.Now())
+	}
+
+	for file := nestedFiles.Front(); file != nil; file = file.Next() {
+		attr := internal.CreateObjAttr(file.Value.(string), 1024, time.Now())
+		suite.rootAttrCacheItem.insert(attr, true, time.Now())
+	}
+
+}
+
+func (suite *cacheMapTestSuite) TestInsertFileCacheMap() {
+	//create path string in form of test/dir/file
+	path := "a/c1/TestFile.txt"
+	startTime := time.Now()
+	attr := internal.CreateObjAttr(path, 1024, startTime)
+
+	//insert path into suite.rootAttrCacheItem
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	//verify correct values are in cacheMapTree
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(1024, cachedItem.attr.Size)
+	suite.assert.EqualValues(false, cachedItem.attr.IsDir())
+}
+
+func (suite *cacheMapTestSuite) TestInsertFolderCacheMap() {
+	//create path string in form of test/dir/file
+	path := "a/c1/TestFolder"
+	startTime := time.Now()
+	attr := internal.CreateObjAttrDir(path)
+
+	//insert path into suite.rootAttrCacheItem
+
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	//verify correct values are in cacheMapTree
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(4096, cachedItem.attr.Size)
+	suite.assert.EqualValues(true, cachedItem.attr.IsDir())
+}
+
+func (suite *cacheMapTestSuite) TestInsertDirsAndFiles() {
+	path := "a/c1/c12/c3p0/newfile.txt"
+	startTime := time.Now()
+	attr := internal.CreateObjAttr(path, 1024, startTime)
+
+	suite.rootAttrCacheItem.insert(attr, true, startTime)
+
+	cachedItem, err := suite.rootAttrCacheItem.get(path)
+
+	suite.assert.Nil(err)
+	suite.assert.NotNil(cachedItem)
+	suite.assert.EqualValues(path, cachedItem.attr.Path)
+	suite.assert.EqualValues(1024, cachedItem.attr.Size)
```

Comment:
```suggestion
	suite.assert.Equal(path, cachedItem.attr.Path)
	suite.assert.Equal(1024, cachedItem.attr.Size)
```
`EqualValues()` is really for objects (pointers). For primitives types (including string!), we should use `Equal()` instead.
This needs to be corrected throughout the tests in this PR.